pages/en/lb3/Authentication-authorization-and-permissions.md

```diff
@@ -45,15 +35,196 @@ LoopBack's access control system is built around a few core concepts, as summari
+Such circumstances require the application to actually manage these different types of users in separated models.
+
+##### Setup
+
+In order to allow the Loopback access control system to work with several models extending the built-in `User` model. The relations between the `users` models and the `AccessToken` should be modified to allow a single `AccessToken` model to host access tokens for multiple types of users while at the same time allowing each `user` models instances to be linked to their related access tokens in a non ambiguous way.  
+
+This is achieved by changing the **hasMany** relation from `User` to `AccessToken` and the **belongsTo** relation from `AccessToken` to `User` by their [polymorphic](Polymorphic-relations) equivalents, in which the `principalType` property is used as a **_discriminator_** to resolve which of the potential `user` model instance an 'accessToken' instance belongs to.
+
+<br>
+{% include note.html content="
+Adapt the following configuration snippets in your custom <code>users</code> and <code>accessToken</code> model definitions.
+"%}
+
+{% include code-caption.html content="/common/models/any-custom-user.json" %}
+```json
+...
+"relations": {
+  "accessTokens": {
+    "type": "hasMany",
+    "model": "customAccessToken",
+    "polymorphic": {
+      "foreignKey": "userId",
+      "discriminator": "principalType"
+    },
+    "options": {
+      "disableInclude": true
+    }
+  }
+},
+...
+```
+
+{% include code-caption.html content="/common/models/custom-access-token.json" %}
+```json
+...
+"relations": {
+  "user": {
+  "type": "belongsTo",
+    "idName": "id",
+    "polymorphic": {
+      "idType": "string",
+      "foreignKey": "userId",
+      "discriminator": "principalType"
+    }
+  }
+},
+...
+```
+
+{% include important.html content="
+In particular, pay attention to:  
+<ul>
+  <li>the <code>model</code> name used to refer to the access token model in the different user models (here named \"customAccessToken\")</li>
+  <li>the <code>idName</code> used for the foreignKey in the access token model referring to the user instance (here named \"id\")</li>
+  <li>the <code>idType</code> used for this foreignKey, according to the type of connector used for the related user models (here using \"string\" for a MongoDB connector for example)</li>
+  <li>use <b>\"principalType\"</b> for the <code>discriminator</code> name, this is mandatory and cannot be changed</li>
+</ul>
+" %}
+
+<br>
+**Last but not least**, do not forget to tell Loopback to use the newly defined custom `accessToken` model by including these lines in the `server.js` file or in a boot script, once again paying attention to the _name_ of the custom `accessToken` model.
+
+```javascript
+var loopback = require('loopback');
+...
+app.use(loopback.token({
+  model: app.models.customAccessToken
+}));
```

Comment:
The idiomatic way is to configure middleware in `server/middleware.json`.

See https://github.com/strongloop/loopback-workspace/blob/fc4f761ed7b635cf18e091b6ad111c31c78880a9/templates/projects/empty-server/files/server/middleware.json#L34

```json
{
  "auth": {
    "loopback#token": {
      "params": {
        "model": "customAccessToken"
      }
    }
  }
}
```

(I typed the example from my head, my not work out of the box.)